src/app/loginComponents/accounts/accounts.component.ts

```diff
@@ -1,12 +1,53 @@
 import { Component, OnInit } from '@angular/core';
-import { Subject } from 'rxjs';
+import { Location } from '@angular/common';
+import { MatTabChangeEvent } from '@angular/material';
+import { FormControl } from '@angular/forms';
+import { Router } from '@angular/router';
 
 @Component({
   selector: 'app-accounts',
   templateUrl: './accounts.component.html'
 })
 export class AccountsComponent implements OnInit {
+  public currentTab = 'accounts'; // default to the 'accounts' tab
+  protected selected = new FormControl(0);
+  validTabs = ['accounts', 'profiles', 'dockstore account controls', 'requests'];
+  constructor(private location: Location, private router: Router) {}
+
   ngOnInit() {
     localStorage.setItem('page', '/accounts');
+    this.parseURL(this.router.url);
+  }
+
+  private parseURL(url: String): void {
+    const decodedUrl: string = decodeURIComponent(url.toString()); // decode any encoded spaces, etc. in the string
```

Comment:
`url` is already a string; it's redundant to call `toString` again, if you go this route.

But I would suggest instead letting built-in JavaScript classes handle this instead of doing your own parsing with Regex

```
const u = new URL(url);
const tab = u.searchParams.get('tab');
...
```

Reply:
I was running into issues with it being a `String` not a `string`, and `toString` seemed to fix that. Will use `searchParams` instead

Reply:
I missed that -- in TypeScript you should be declaring your strings as `string`, not `String`.

Reply:
Oh okay